From 01670f1641b23deb84bb95f855eacb4c8c0d66d6 Mon Sep 17 00:00:00 2001
From: Olivier Blin <olivier.blin@softathome.com>
Date: Fri, 17 Nov 2023 09:52:35 -0800
Subject: [PATCH] Fix AudioSourceProviderGStreamer build with
 !ENABLE(MEDIA_STREAM) https://bugs.webkit.org/show_bug.cgi?id=265039

Reviewed by Philippe Normand.

* Source/WebCore/platform/audio/gstreamer/AudioSourceProviderGStreamer.h:
WebAudioSourceProvider.h always needs to be included since
AudioSourceProviderGStreamer inherits from WebAudioSourceProvider.

Canonical link: https://commits.webkit.org/270898@main

Upstream-Status: Accepted [https://bugs.webkit.org/show_bug.cgi?id=265039]
---
 .../platform/audio/gstreamer/AudioSourceProviderGStreamer.h     | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Source/WebCore/platform/audio/gstreamer/AudioSourceProviderGStreamer.h b/Source/WebCore/platform/audio/gstreamer/AudioSourceProviderGStreamer.h
index be04997c..1a8f2ddb 100644
--- a/Source/WebCore/platform/audio/gstreamer/AudioSourceProviderGStreamer.h
+++ b/Source/WebCore/platform/audio/gstreamer/AudioSourceProviderGStreamer.h
@@ -24,6 +24,7 @@
 #include "AudioSourceProviderClient.h"
 #include "GRefPtrGStreamer.h"
 #include "MainThreadNotifier.h"
+#include "WebAudioSourceProvider.h"
 #include <gst/gst.h>
 #include <wtf/Forward.h>
 #include <wtf/Lock.h>
@@ -32,7 +33,6 @@
 #if ENABLE(MEDIA_STREAM)
 #include "GStreamerAudioStreamDescription.h"
 #include "MediaStreamTrackPrivate.h"
-#include "WebAudioSourceProvider.h"
 #endif
 
 typedef struct _GstAdapter GstAdapter;
-- 
2.30.2

